R/Seurat.R

```diff
@@ -149,33 +170,31 @@ from_Seurat <- function(seurat_obj, output_class = c("InMemoryAnnData", "HDF5Ann
     ...
   )
 
-  # trackstatus: class=Seurat, feature=set_X, status=wip
-  # trackstatus: class=Seurat, feature=set_layers, status=wip
-  for (assay_name in names(seurat_obj@assays)) {
-    # TODO: Maybe we shouldn't use counts but instead data
-    assay_data <- SeuratObject::GetAssayData(seurat_obj, "counts", assay = assay_name)
-
-    if (nrow(assay_data) != length(var_names) || !identical(rownames(assay_data), var_names)) {
-      warning(
-        "Skipping assay '", assay_name, "' because it has different feature names ",
-        "than the active assay ('", seurat_obj@active.assay, "')."
-      )
-      next
-    }
-    if (ncol(assay_data) != length(obs_names) || !identical(colnames(assay_data), obs_names)) {
-      warning(
-        "Skipping assay '", assay_name, "' because it has different cell names ",
-        "than the active assay ('", seurat_obj@active.assay, "')."
-      )
-      next
+  if (!is.null(X)) {
+    # Check if the slot is not empty
+    if (all(dim(SeuratObject::GetAssayData(seurat_obj, slot = X, assay = assay_name)) == 0)) {
+      stop("The '", X, "' slot is empty.")
     }
 
-    # remove names
+    assay_data <- SeuratObject::GetAssayData(seurat_obj, slot = X, assay = assay_name)
+
+    # Remove names
     dimnames(assay_data) <- list(NULL, NULL)
-    if (assay_name == seurat_obj@active.assay) {
-      ad$X <- Matrix::t(assay_data)
-    } else {
-      ad$layers[[assay_name]] <- Matrix::t(assay_data)
+    ad$X <- Matrix::t(assay_data)
+  } else {
+    # Cannot compare other values with NULL
+    X <- "none"
+  }
+
+  # Add the remaining non-empty slots as layers
+  slots <- c("counts", "data", "scale.data")
+  slots <- slots[slots != X]
+
+  for (slot in slots) {
+    if (!all(dim(GetAssayData(seurat_obj, slot = slot)) == 0)) {
```

Comment:
GetAssayData is not imported

```suggestion
    if (!all(dim(SeuratObject::GetAssayData(seurat_obj, slot = slot)) == 0)) {
```